### DISCUSSION TITLE
Can I use scientific notation for spectral data?

### DISCUSSION THREAD

**Discussion by ademmler** [colour-science/colour#917] [Q&A] [answered]
Hi everybody,

from my measurement device I get spectral data like this:
360.00  4.9591E-002
 361.00  6.0577E-002
 362.00  9.0296E-002
 363.00  1.5385E-001
 364.00  2.4088E-001
 365.00  3.1027E-001
....

Can I use this directly to describe a spectrum? For me it seems not to be possible.
If not - is there an exiting "conversion function" to preferred "Colour" formats?
What is the best practise?

thx in advance
Alexander

**Comment by thomasmansencal** [comment, marked as answer]
Hi @ademmler,

Yes this should be easily possible:

```python
import colour
import numpy as np
data = """
360.00 4.9591E-002
361.00 6.0577E-002
362.00 9.0296E-002
363.00 1.5385E-001
364.00 2.4088E-001
365.00 3.1027E-001
""".strip().split('\n')
data = np.loadtxt(data, delimiter=' ')
colour.SpectralDistribution(data[..., 1], data[..., 0])
Out[6]: 
SpectralDistribution([[  3.60000000e+02,   4.95910000e-02],
                      [  3.61000000e+02,   6.05770000e-02],
                      [  3.62000000e+02,   9.02960000e-02],
                      [  3.63000000e+02,   1.53850000e-01],
                      [  3.64000000e+02,   2.40880000e-01],
                      [  3.65000000e+02,   3.10270000e-01]],
                     interpolator=SpragueInterpolator,
                     interpolator_kwargs={},
                     extrapolator=Extrapolator,
                     extrapolator_kwargs={'method': 'Constant', 'left': None, 'right': None})
```

**Reply by ademmler** [reply]
Hi, thx for helping me with this.

Yes this does the trick. I can use this to plot the spectrum.
sd = colour.SpectralDistribution(data[..., 1], data[..., 0])   
plot_single_sd(sd)

But how did you got the output starting at Out[6]: ?

**Reply by thomasmansencal** [reply]
It is the default output when you `print` a `colour.SpectralDistribution`  class instance, being in the REPL, it got printed out by default.

**Reply by ademmler** [reply]
Thx for your response. May I do not understand fully.
If I do this I get less output compared to you.

my commands:
sd = colour.SpectralDistribution(data[..., 1], data[..., 0])
print(sd)

Output:
[[  3.50000000e+02   3.56770000e-03]
 [  3.51000000e+02   1.66280000e-02]
 [  3.52000000e+02   8.34640000e-03]
 [  3.53000000e+02   0.00000000e+00]
 [  3.54000000e+02   0.00000000e+00]
 [  3.55000000e+02   0.00000000e+00]
 [  3.56000000e+02   0.00000000e+00]
 [  3.57000000e+02   0.00000000e+00]
 [  3.58000000e+02   0.00000000e+00]
 [  3.59000000e+02   1.51500000e-02]
 [  3.60000000e+02   4.95910000e-02]]

**Reply by thomasmansencal** [reply]
Try `repr(sd)` which is what IPython uses under the hood instead of `print`: https://www.codingem.com/str-vs-repr-in-python/

**Reply by ademmler** [reply]
Thx for making this clear - very helpful.
